src/index/CompressedRelation.cpp

```diff
@@ -874,13 +874,17 @@ void CompressedRelationWriter::compressAndWriteBlock(
     AD_CORRECTNESS_CHECK(lastCol0Id == last[0]);
 
     auto [hasDuplicates, graphInfo] = getGraphInfo(block);
+    // The block indices will be set later, write a recognizable dummy value for
+    // easier debugging.
+    static constexpr size_t blockIndexDummy = 4387;
```

Comment:
Is there a good reason why we can't know the block index when the block is written? Maybe because the blocks are written in parallel? Anyway, if there is a good reason, we should define a constant with a value like `std::numeric_limits<size_t>::max()` and a meaningful name like `BLOCK_INDEX_NOT_YET_SET` or something like that.

Reply:
Yes, the parallelism is the issue here, I have added a comment that explains that and made the constant look more like something that looks intentions.
I don't like `max()`  as it is equal to `-1` which often occurs by accidental bugs.

Reply:
Well, you could take `max() - 1` or `max() - 2` and you could certainly add a static constant class member `BLOCK_INDEX_NOT_YET_SET`  to give this thing a proper name in the code and in the tests.